### PR TITLE
Availability Web

### PR DESCRIPTION
**Pull request by sjmcculloch**
New web addition built on material-ui containing:

* Apollo 2 Integration (still need to reset store on logout)

* Create Request

image

* View Requests

image

* View Request Detail

image

---

Still to add:

- Inline editing of new/existing availability timeSegments
- View by Day
- View by User
- Grid Filters and additional meta information
- Capabilities

### CODE REVIEW

**Comment by sdunster** [comment]
#micdrop 
this might take some time to review..

File: web/src/screens/private/Schedules/AddSchedule.js [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+import CURRENT_USER_QUERY from '../../../graphql/current-user.query';
+
+import styles from './AddSchedule.styles';
+
+function getSteps() {
+  return ['Select group', 'Specify details', 'Capabilities', 'Enter dates'];
+}
+
+function getStepContent(step) {
+  switch (step) {
```
Comment (sdunster) [inline web/src/screens/private/Schedules/AddSchedule.js]: suggest:
```
const steps = [
  'Who will the request target?',
  '...',
  '...',
];
if(step >= steps.length) {
  return 'Unknown step';
}
return steps[step];
```

File: web/src/screens/private/Schedules/AddSchedule.js [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+  onUseDatesChange = () => {
+    const useDates = !this.state.useDates;
+    this.setState(() => ({ useDates }));
+  };
+
+  handleNext = () => {
+    const steps = getSteps();
+    if (this.state.activeStep === steps.length - 1) {
+      const { name, details, useDates, startTime, endTime, groupId } = this.state;
+      let startTimeUnix = 0;
```
Comment (sdunster) [inline web/src/screens/private/Schedules/AddSchedule.js]: i'd really prefer these were constants somewhere even if it's not the same file as the client/server.

File: web/src/screens/private/Schedules/AddSchedule.js [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+            <div style={{ display: 'flex' }}>
+              <NavLink to="/dashboard">
+                <ArrowBackIcon className={classes.cardIcon} />
+              </NavLink>
+              <Typography variant="title" color="inherit" className={classes.cardTitle}>
+                Add New Request
+              </Typography>
+            </div>
+            <Stepper activeStep={activeStep} orientation="vertical">
+              {steps.map((label, index) => (
```
Comment (sdunster) [inline web/src/screens/private/Schedules/AddSchedule.js]: not important: I feel like putting these in separate components or something might not be a terrible idea. This function/JSX block is massive.

File: web/src/screens/public/LoginPage.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    this.setState(() => ({ username }));
+  };
+  onPasswordChange = (e) => {
+    const password = e.target.value;
+    this.setState(() => ({ password }));
+  };
+
+  onSubmit = (e) => {
+    e.preventDefault();
+    const deviceUuid = uuidv4();
```
Comment (sdunster) [inline web/src/screens/public/LoginPage.js]: hm this brings about an interesting discussion. Should we treat browsers as devices? I think yes. We could even send push notifications to chrome and/or safari using the same frameworks as iOS and/or Android respectively.

File: server/src/main.js [hunk 5]
```diff
@@ -11,6 +11,8 @@ import { getHandlers } from './logic';
 import { getResolvers } from './resolvers';
 import { getCreators } from './creators';
 
+const cors = require('cors');
```
[the remark this answers is not in the document]
Comment (sjmcculloch) [reply server/src/main.js]: cars ?

File: server/src/main.js [hunk 6]
```diff
@@ -37,6 +41,7 @@ app.use(
   graphqlExpress((req) => {
     let user;
     let device;
+    console.log(req);
```
[the remark this answers is not in the document]
Comment (sjmcculloch) [reply server/src/main.js]: removed

File: web/package.json [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  "name": "callout-web",
+  "version": "0.0.1",
+  "private": true,
+  "scripts": {
+    "build:dev": "webpack",
+    "build:prod": "webpack -p --env production",
+    "dev-server": "webpack-dev-server",
+    "lint": "./node_modules/.bin/eslint src",
+    "test": "jest --config=jest.config.json",
+    "start": "node server/server.js",
```
[the remark this answers is not in the document]
Comment (sjmcculloch) [reply web/package.json]: Client testing of full production server - can be ignored and/or removed.

File: web/server/server.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+const path = require('path');
```
[the remark this answers is not in the document]
Comment (sjmcculloch) [reply web/server/server.js]: Yep

File: web/src/routers/AppRouter.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+import Header from '../screens/partial/Header';
+
+import PrivateRoute from './PrivateRoute';
+import PublicRoute from './PublicRoute';
+
+const AppRouter = () => (
+  <BrowserRouter>
+    <div>
+      <Reboot />
```
[the remark this answers is not in the document]
Comment (sjmcculloch) [reply web/src/routers/AppRouter.js]: It's a reset for CSS.

[the remark this answers is not in the document]
Comment (sjmcculloch) [reply web/src/routers/AppRouter.js]: naming changes in next version as reboot keyword not the best.

File: web/src/screens/private/Schedules/AddSchedule.js (hunk 1 above)
Comment (sjmcculloch) [reply web/src/screens/private/Schedules/AddSchedule.js]: Much better - thank you !

File: web/src/screens/private/Schedules/AddSchedule.js (hunk 2 above)
Comment (sjmcculloch) [reply web/src/screens/private/Schedules/AddSchedule.js]: I agree - I'll fix up

File: web/src/screens/private/Schedules/AddSchedule.js (hunk 3 above)
Comment (sjmcculloch) [reply web/src/screens/private/Schedules/AddSchedule.js]: Yes I agree - just worked on getting it worker and then I'll split what I can into smaller components.

File: web/src/screens/public/LoginPage.js (hunk 4 above)
Comment (sjmcculloch) [reply web/src/screens/public/LoginPage.js]: I agree - yes

File: server/src/main.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (sdunster) [reply server/src/main.js]: lol whoops

**Pull request merged**